mu4e/mu4e-compose.el

```diff
@@ -386,33 +457,13 @@ removing the In-Reply-To header."
           (define-key map (kbd "C-S-u")   'mu4e-update-mail-and-index)
           (define-key map (kbd "C-c C-u") 'mu4e-update-mail-and-index)
           (define-key map (kbd "C-c C-k") 'mu4e-message-kill-buffer)
-          (define-key map (kbd "M-q")     'mu4e-fill-paragraph)
+          (if mu4e-compose-format-flowed
+              (define-key map (kbd "C-c C-c" 'mu4e-fill-send-and-exit)))
```

Comment:
Hmm, this would make things depend on the value of `mu4e-compose-format-flowed` the first time you use it (with no obvious way to change it later) - or am i misreading?

Also, could we keep `M-q` for filling too? That has the advantage that you can see the result.


Overall, I think what'd I'd prefer is having `mu4e-compose-fill-policy` defvar, with values nil, 'always, 'ask, and do the filling in `send-mode-hook`. 

What do you think?

Reply:
> Overall, I think what'd I'd prefer is having mu4e-compose-fill-policy defvar, with values nil, 'always, 'ask, and do the filling in send-mode-hook.

We can definitely do that.  Here's my use-case, though: I would like to send ~90% of my emails with a format=flowed line length of 72, since that follows the spec and is the only one *guaranteed* not to display obnoxiously long lines in terminal clients.  However, for about 10% of my emails (primarily ones I send specifically to non-technical recipients/when I know they will be viewed on a cell phone) I'd like to set line length to 998 to take advantage of the [line-wrapping behavior](https://vxlabs.com/2019/08/25/format-flowed-with-long-lines/) of clients that don't implement format=flowed.

Given that use case, I wouldn't want mu4e to ask about line length _every_ time, since it's the same 90% of the time.  But I also would like a way to get it to ask when I _do_ want to set the length.  Accepting a prefix argument works very well for that.

I don't want to insist on implementing my use case or anything – I can always accommodate it my own `.init` file.  But I thought I'd mention it, since it seems (to me at least) like others might also like being able to change the behavior via a prefix argument.

How about this modification to your suggestion: bind `C-c C-c` to a new function `mu4e-send-and-exit`.  Without a prefix argument (or when`mu4e-compose-format-flowed` is nil), that function calls `message-send-and-exit`, but when called with a prefix argument (with `mu4e-compose-format-flowed` non nil), it prompts for a line length.  That keeps the prefix argument support that strikes me as useful, but also lets us use the `mu4e-compose-fill-policy` var and the `message-send-hook`.  (I assume that's what you meant by `send-mode-hook`, but let me know if there's a separate `send-mode-hook` I'm missing.)

Or, if you prefer, I'm happy to write it as you suggested and implement my own workflow in my `init` file.

> Also, could we keep M-q for filling too? That has the advantage that you can see the result.

That's a good point.  The reason I didn't bind `M-q` to `mu4e-fill-buffer-for-flowed` was to avoid shadowing `fill-paragraph`.  That wasn't a problem before – in fact, it was _good_ to shadow `fill-paragraph` when users were required to write message paragraphs as a single line, since using the regular `fill-paragraph` would break that.  But under this PR, users might use `fill-paragraph` to make their draft formatting better.

What if we bind `mu4e-fill-buffer-for-flowed` to `C-c M-q` and leave `M-q` free for `fill-paragraph`?